pcsx2/gui/Debugger/CtrlRegisterList.cpp

```diff
@@ -20,505 +20,436 @@
-		}
-
-		int x = 17+(maxLen+3)*charWidth;
-		startPositions.push_back(x);
-		currentRows.push_back(0);
-	}
-
-	SetDoubleBuffered(true);
-	SetInitialSize(ClientToWindowSize(GetMinClientSize()));
+    wxGrid* regGrid = registerGrids[cat];
+    int numRows = regGrid->GetNumberRows();
+    int numCols = regGrid->GetNumberCols();
+
+    std::vector<ChangedReg>& changedRegs = changedCategories[cat];
+    DebugInterface::RegisterType type = cpu->getRegisterType(cat);
+    for (int row = 0; row < numRows; row++)
+    {
+        wxColor bgColor = regGrid->GetGridCursorRow() == row ? wxColor(0xFFFFDFE0) :
+                                                               wxColor(0xFFFFEFFF);
+        u128 value = cpu->getRegister(cat, row);
+        ChangedReg& changed = changedRegs[row];
+
+        for (int col = 0; col < numCols; col++)
+        {
+            const wxColor colorChanged = wxColor(0xFF0000FF);
+            const wxColor colorUnchanged = wxColor(0xFF004000);
+            wxColor textColor;
+            wxString cellValue;
+
+            switch (type)
+            {
+            case DebugInterface::NORMAL:
+                cellValue = wxString::Format("%08X", value._u32[numCols - col - 1]);
+                textColor = changed.changed[numCols - col - 1] ? colorChanged : colorUnchanged;
+                break;
+            case DebugInterface::SPECIAL:
+                cellValue = cpu->getRegisterString(cat, row);
+                textColor = (changed.changed[0] || changed.changed[1] || changed.changed[2] || changed.changed[3]) ?
+                    colorChanged : colorUnchanged;
+                break;
+            default: pxAssert(0 && "Unreachable switch case");
+            }
+            if (regGrid->GetCellTextColour(row, col) != textColor)
+                regGrid->SetCellTextColour(row, col, textColor);
+            if (regGrid->GetCellValue(row, col) != cellValue)
+                regGrid->SetCellValue(row, col, cellValue);
+            if (regGrid->GetCellBackgroundColour(row, col) != bgColor)
+                regGrid->SetCellBackgroundColour(row, col, bgColor);
+        }
+    }
+}
+
+void CtrlRegisterList::updateSize(int cat)
+{
+    wxGrid* regGrid = registerGrids[cat];
+
+    // Make sure that the number of shown cols is correct.
+    {
+        int regBits = cpu->getRegisterSize(cat);
+        int numCols = regGrid->GetNumberCols();
+
+        int shownCols = 0;
+        while (shownCols < numCols && regGrid->IsColShown(shownCols)) shownCols++;
+        if (shownCols > maxBits / 32)
+            shownCols = (maxBits / 32);
+        else if (shownCols < regBits / 32)
+            shownCols = std::min(maxBits / 32, regBits / 32);
+
+        for (int col = 0; col < numCols; col++)
+            if (col < shownCols)
+                regGrid->ShowCol(numCols - col - 1); // Big-endian representation so flip order
+            else
+                regGrid->HideCol(numCols - col - 1); // Big-endian representation so flip order
+    }
```

Comment:
I don't think the `{` and `}` are needed, and it seems you've used them for clarity instead of for restricting scope. I think your comments have made the intention clear enough

Same applies to the hacks bit below.